rstudio/rsconnect-python
# add support for deployment on RStudio Cloud

### Description

Adds:
- Support for adding RStudio Cloud credentials (currently token/secret pairs)
  - to specify a credential is for Cloud, pass `--server rstudio.cloud` (or `--server https://api.staging.rstudio.cloud` for staging)
  - if only the token/secret pair is provided, rsconnect-python will default to using shinyapps.io
- Support for deployment to RStudio Cloud
  - Outputs deployed to RStudio Cloud can optionally be associated with existing projects by providing that application's id through the LUCID_APPLICATION_ID variable (or `--project-application-id`). 

### Testing Notes / Validation Steps

We should validate the following cases:
- Deploy to cloud using `rsconnect deploy shiny`
- Deploy to cloud using `rsconnect deploy manifest`
- Deploy to cloud using `rsconnect deploy shiny` and `LUCID_APPLICATION_ID` – the project should show the deployed output in its list of associated outputs
- Deploy to cloud using `rsconnect deploy manifest` and `LUCID_APPLICATION_ID` – the project should show the deployed output in its list of associated outputs
